ui/app/AppLayouts/Profile/Sections/ChangeProfilePicModal.qml

```diff
@@ -0,0 +1,150 @@
+import QtQuick 2.13
+import QtQuick.Dialogs 1.3
+import "../../../../imports"
+import "../../../../shared"
+import "../../../../shared/status"
+
+ModalPopup {
+    property string selectedImage // selectedImage is for us to be able to analyze it before setting it as current
+    property string uploadError
+
+    id: popup
+
+    title: qsTr("Profile picture")
+
+    onClosed: {
+        destroy()
+    }
+
+    onSelectedImageChanged: {
+        if (!selectedImage) {
+            return
+        }
+
+        cropImageModal.open()
+    }
+
+    Item {
+        anchors.fill: parent
+
+        RoundedImage {
+            id: profilePic
+            source: profileModel.profile.largeImage
+            width: 160
+            height: 160
+            anchors.verticalCenter: parent.verticalCenter
+            anchors.horizontalCenter: parent.horizontalCenter
+            border.width: 1
+            border.color: Style.current.border
+            onClicked: imageDialog.open()
+        }
+
+        StyledText {
+            visible: !!uploadError
+            text: uploadError
+            anchors.left: parent.left
+            anchors.right: parent.right
+            anchors.top: profilePic.bottom
+            horizontalAlignment: Text.AlignHCenter
+            font.pixelSize: 13
+            wrapMode: Text.WordWrap
+            anchors.topMargin: 13
+            font.weight: Font.Thin
+            color: Style.current.danger
+        }
+
+        ModalPopup {
+            id: cropImageModal
+            width: image.width + 50
+            height: image.height + 170
+            title: qsTr("Crop your image (optionnal)")
+
+            Image {
+                id: image
+                width: 400
+                source: popup.selectedImage
```

Comment:
Isn't it necessary to specify the height?

Reply:
Nope, when you have an image with `PreserveAspectFit`, if you set the width, the height will adapt to keep the aspect ratio